src/client/content_cao.cpp

```diff
@@ -814,19 +825,21 @@ void GenericCAO::addToScene(ITextureSource *tsrc, scene::ISceneManager *smgr)
 		setBillboardTextureMatrix(m_spritenode, 1, 1, 0, 0);
 
 		// This also serves as fallback for unknown visual types
-		if (m_prop.visual != "sprite") {
-			infostream << "GenericCAO::addToScene(): \"" << m_prop.visual
+		if (m_prop.visual != OBJECTVISUAL_SPRITE) {
+			infostream << "GenericCAO::addToScene(): \""
+				<< enum_to_string(es_ObjectVisual, m_prop.visual)
```

Comment:
well it will always say "unknown" won't it?
I propose to remove this as the deserialize method will log the true string sent by the server.